fortio_main.go

```diff
@@ -536,9 +538,25 @@ func grpcClient() {
 		}
 		return
 	}
-	_, err := fgrpc.PingClientCall(host, count, httpOpts.PayloadUTF8(), *pingDelayFlag, &httpOpts.TLSOptions)
+	_, err := fgrpc.PingClientCall(host, count, httpOpts.PayloadUTF8(), *pingDelayFlag, &httpOpts.TLSOptions, md)
 	if err != nil {
 		// already logged
 		os.Exit(1)
 	}
 }
+
+// httpHeader2grpcMetadata covert md's key to lowercase and filter invalid key.
+func httpHeader2grpcMetadata(headers map[string][]string) map[string][]string {
+	ret := make(map[string][]string)
+	for k, v := range headers {
+		k = strings.ToLower(k)
```

Comment:
so http headers are automatically (when using Get for instance) case insensitive - it's not true/the case for grpc meta data? the grpc lib doesn't automatically set that lowercase for us?

Reply:
I checked indeed if you don't you get
```
14:53:17 W grpcrunner.go:109> Error making grpc call: rpc error: code = Internal desc = header key "Foo" contains illegal characters not in [0-9a-z-_.]
```

Reply:
(sorry for late reply) 
As I mentioned above
https://github.com/fortio/fortio/pull/671#discussion_r1062558141
There _may_ be some problems with grpc-go's medata validation,
https://github.com/grpc/grpc-go/issues/5913
so I think we should ensure the validity of the metadata ourselves instead of relying on external conversions